python/ray/dag/tests/experimental/test_torch_tensor_dag.py

```diff
@@ -1172,6 +1172,362 @@ def destroy(self) -> None:
+            dag_node = compiled_dag.idx_to_task[exec_task.task_idx].dag_node
+            if isinstance(dag_node, CollectiveOutputNode):
+                assert exec_task.collective_group is not None
+                nccl_group_id = exec_task.collective_group._type_hint._nccl_group_id
+                assert nccl_group_id is not None
+                comms.add(nccl_group_id)
+    # Only 1 NCCL group should be created.
+    assert len(comms) == 1
+    comm = list(comms)[0]
+    # The comm for all-reduce should be the same as the p2p send/recv comm.
+    assert comm == compiled_dag._nccl_group_id
+
+    # Sanity check: the compiled dag can execute.
+    shape = (10,)
+    dtype = torch.float16
+    ref = compiled_dag.execute([(shape, dtype, i + 1) for i in range(num_workers)])
+    result = ray.get(ref)
+    reduced_val = (num_workers + 1) * num_workers / 2
+    assert result == [(reduced_val, shape, dtype) for _ in workers]
+
+    compiled_dag.teardown()
+
+    receiver = workers[1]
+    with InputNode() as inp:
+        computes = [
+            worker.compute_with_tuple_args.bind(inp, i)
+            for i, worker in enumerate(workers)
+        ]
+        collectives = collective.allreduce.bind(computes)
+        dag = receiver.recv.bind(
+            collectives[0].with_type_hint(TorchTensorType(transport="nccl"))
+        )
+
+    compiled_dag = dag.experimental_compile()
+
+    comms = set()
+    for _, exec_tasks in compiled_dag.actor_to_executable_tasks.items():
+        for exec_task in exec_tasks:
+            dag_node = compiled_dag.idx_to_task[exec_task.task_idx].dag_node
+            if isinstance(dag_node, CollectiveOutputNode):
+                assert exec_task.collective_group is not None
+                nccl_group_id = exec_task.collective_group._type_hint._nccl_group_id
+                assert nccl_group_id is not None
+                comms.add(nccl_group_id)
+
+    # Both workers participated in the all-reduce;
+    # one of them is the sender and the other is the receiver in p2p.
+    # So only 1 NCCL group should be created.
+    assert len(comms) == 1
+    # The following assertion fails b/c only the downstream node is added to
+    # p2p nccl group when transport=nccl is specified.
+    # comm = list(comms)[0]
+    # # The comm for all-reduce should be the same as the p2p send/recv comm.
+    # assert comm == compiled_dag._nccl_group_id
+
+    # Sanity check: the compiled dag can execute.
+    shape = (10,)
+    dtype = torch.float16
+    ref = compiled_dag.execute([(shape, dtype, i + 1) for i in range(num_workers)])
+    result = ray.get(ref)
+    reduced_val = (num_workers + 1) * num_workers / 2
+    assert result == (reduced_val, shape, dtype)
+
+    compiled_dag.teardown()
+
+
+@pytest.mark.parametrize("ray_start_regular", [{"num_cpus": 4}], indirect=True)
+def test_torch_tensor_nccl_all_reduce_diff_comms(ray_start_regular):
+    """
+    Test communicators are deduplicated when the collective and the p2p send/recv
```

Comment:
Can you update the comment to match the test?

Reply:
Ack